metrics-exporter-prometheus/src/formatting.rs

```diff
@@ -110,8 +110,17 @@ pub fn write_metric_line<T, T2>(
 /// [data model]: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
 pub fn sanitize_metric_name(name: &str) -> String {
     // The first character must be [a-zA-Z_:], and all subsequent characters must be [a-zA-Z0-9_:].
-    name.replacen(invalid_metric_name_start_character, "_", 1)
-        .replace(invalid_metric_name_character, "_")
+    let mut out = String::with_capacity(name.len());
+    let mut is_invalid: fn(char) -> bool = invalid_metric_name_start_character;
+    for c in name.chars() {
+        if is_invalid(c) {
+            out.push('_');
+        } else {
+            out.push(c);
+        }
+        is_invalid = invalid_metric_name_character;
+    }
+    return out;
```

Comment:
No need for the return here, as the last expression in scope will be used as the return value.

```suggestion
    out
```